src/lib.rs

```diff
@@ -347,6 +347,43 @@ impl RecoverableSignature {
         (RecoveryId(recid), ret)
     }
 
+    #[inline]
+    /// Converts a compact-encoded byte slice from Bitcoin Core to a recoverable signature.
+    pub fn from_bitcoin_compact<C>(
+        secp: &Secp256k1<C>,
+        data: &[u8],
+    ) -> Result<RecoverableSignature, Error> {
+        if data.len() != 65 {
+            return Err(Error::InvalidSignature);
+        }
+
+        // Bitcoin Core sets the first byte to `27 + rec + (fCompressed ? 4 : 0)`.
+        let rec_id = RecoveryId::from_i32(if data[0] >= 31 {
+            (data[0] - 31) as i32
+        } else {
+            (data[0] - 27) as i32
+        })?;
+        RecoverableSignature::from_compact(secp, &data[1..], rec_id)
+    }
+
+    #[inline]
+    /// Serializes the recoverable signature to Bitcoin Core-compatible compact format.
+    pub fn serialize_bitcoin_compact<C>(&self, secp: &Secp256k1<C>) -> [u8; 65] {
+        let mut ret = [0u8; 65];
+        let mut recid = 0i32;
+        unsafe {
+            let err = ffi::secp256k1_ecdsa_recoverable_signature_serialize_compact(
+                secp.ctx,
+                ret[1..].as_mut_ptr(),
+                &mut recid,
+                self.as_ptr(),
+            );
+            assert!(err == 1);
+        }
+        ret[0] = (27 + 4 + recid) as u8; // compressed
```

Comment:
This is a bit of a problem. It looks like Bitcoin Core sets the compressed flag based on the public key, which we don't have here of course.